### PR TITLE
MODSOURCE-707 Handle authority domain delete events

### PR DESCRIPTION
**Pull request by psmagin**
## Purpose
This story focuses on the consumption of authority domain events within the mod-source-record-storage module. It involves processing these events based on specific conditions. On a soft-delete event, the corresponding record will be marked with the "DELETED" state. On a hard-delete event, the record will be completely deleted from the database. Events of other domain types will be ignored.

## Approach
- create a new verticle for consumption of domain events (verticles for consumers were refactored, common logic moved to absract class)
- create event hadler
- removed scheduled removing of records with DELETE state

## Learning
https://issues.folio.org/browse/MODSOURCE-707

### CODE REVIEW

File: mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  public AuthorityDomainKafkaHandler(RecordService recordService) {
+    this.recordService = recordService;
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> consumerRecord) {
+    log.trace("handle:: Handling kafka record: '{}'", consumerRecord);
+    String authorityId = consumerRecord.key();
+    if (isUnexpectedDomainEvent(consumerRecord)) {
+      log.trace("handle:: Expected only {} domain type. Skipping authority domain kafka record [ID: '{}']",
```
Comment (okolawole-ebsco) [inline mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java:39]: it the domain event is unexpected, should a `return` be here?

File: mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          return Future.succeededFuture();
+        }
+        var matchedId = recordCollection.getRecords().get(0).getMatchedId();
+
+        return recordService.updateRecordsState(matchedId, RecordState.DELETED, RecordType.MARC_AUTHORITY, tenantId);
+      }).map(authorityId);
+  }
+
+  private Future<String> performHardDelete(String authorityId, String tenantId) {
+    return recordService.deleteRecordsByExternalId(authorityId, tenantId).map(authorityId);
```
Comment (okolawole-ebsco) [inline mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java:70]: is authority id and external Id the same thing?

File: mod-source-record-storage-server/src/main/java/org/folio/verticle/consumers/AbstractConsumerVerticle.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      subscriptionDefinition = SubscriptionDefinition.builder()
+        .eventType(eventType)
+        .subscriptionPattern(createSubscriptionPattern(kafkaConfig.getEnvId(), eventType))
+        .build();
+    }
+    return subscriptionDefinition;
+  }
+
+  private String getConsumerName() {
+    return constructModuleName() + "_" + getClass().getSimpleName();
```
Comment (okolawole-ebsco) [inline mod-source-record-storage-server/src/main/java/org/folio/verticle/consumers/AbstractConsumerVerticle.java:90]: Does this include the module version?

Comment (psmagin) [reply mod-source-record-storage-server/src/main/java/org/folio/verticle/consumers/AbstractConsumerVerticle.java:90]: Yes

File: mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java (hunk 2 above)
Comment (psmagin) [reply mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java:70]: Yes

File: mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java (hunk 1 above)
Comment (psmagin) [reply mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java:39]: Yes, fixed.

File: mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java (hunk 2 above)
Comment (okolawole-ebsco) [reply mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java:71]: Is it possible for an authority to have an external id that is not the authority id?

**Review by okolawole-ebsco — APPROVED** [review]

File: mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java (hunk 2 above)
Comment (psmagin) [reply mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java:71]: I don't think such case exists.

**Review by TsaghikKhachatryan — APPROVED** [review]

**Review by viacheslavkol — APPROVED** [review]

**Pull request merged**